SoftLayer/CLI/hardware/create_credential.py

```diff
@@ -0,0 +1,57 @@
+"""Create a password for a software component"""
+# :license: MIT, see LICENSE for more details.
+
+import click
+
+import SoftLayer
+from SoftLayer.CLI import environment
+from SoftLayer.CLI import formatting
+
+
+@click.command(cls=SoftLayer.CLI.command.SLCommand, )
+@click.argument('identifier')
+@click.option('--username', '-U', help="The username part of the username/password pair")
+@click.option('--password', '-P', help="The password part of the username/password pair.")
+@click.option('--notes', '-n', help="A note string stored for this username/password pair.")
+@click.option('--system', help="The name of this specific piece of software.")
+@environment.pass_env
+def cli(env, identifier, username, password, notes, system):
+    """Create a password for a software component."""
+
+    mgr = SoftLayer.HardwareManager(env.client)
+
+    software = mgr.get_software_components(identifier)
+    sw_id = None
+    for sw_instance in software:
+        if (sw_instance['softwareLicense']['softwareDescription']['name']).lower() == system:
+            sw_id = sw_instance['id']
+            break
+        elif system:
+            if (sw_instance['softwareLicense']['softwareDescription']['name']) == 'Passmark Suite':
+                sw_id = sw_instance['id']
```

Comment:
I dont understand what you are checking for here?

Looks to me like you are checking for if a user picked `--system`, but then you just set sw_id to `Passmark Suite`. I feel like this whole `elif` check could be removed, and if at the end if your for loop, if `sw_id` is None and `system` is not None, throw and exception about not being able to find a software called `system`